rocks-lib/src/build/mod.rs

```diff
@@ -32,7 +33,32 @@ fn install(
     Ok(())
 }
 
-pub fn build(rockspec: Rockspec, config: &Config) -> Result<()> {
+#[async_recursion]
+pub async fn build(rockspec: Rockspec, config: &Config) -> Result<()> {
+    // TODO(vhyrro): Create a unified way of accessing the Lua version with centralized error
+    // handling.
+    let lua_version = rockspec.lua_version();
+    let lua_version = config.lua_version().or(lua_version.as_ref()).ok_or_eyre(
+        "lua version not set! Please provide a version through `--lua-version <ver>`",
+    )?;
+
+    let tree = Tree::new(config.tree().clone(), lua_version.clone())?;
+
+    // Recursively build all dependencies.
```

Comment:
thought: we're going to have to implement an equivalent to the luarocks loader to support multiple versions of the same dependency.
And for people who don't have the loader enabled, we might want to consider adding a flag that disallows conflicting dependencies. Which would mean we'll need a more sophisticated dependency resolution.

Reply:
creating an issue for this so that we can get this merged

Reply:
This could be a bit of metadata that is read from the tree's lockfile in the future (e.g. a dependency or rock could have an `only` or `singleton` flag attached). Good point though.